erts/emulator/nifs/win32/win_prim_file.c

```diff
@@ -491,6 +491,42 @@ posix_errno_t efile_open(const efile_path_t *path, enum efile_modes_t modes,
     }
 }
 
+posix_errno_t efile_from_fd(int fd,
+                            enum efile_modes_t modes,
+                            ErlNifResourceType *nif_type,
+                            efile_data_t **d) {
+    HANDLE handle;
+
+    handle = (HANDLE)_get_osfhandle(fd);
+
+    if(handle != INVALID_HANDLE_VALUE) {
+        efile_win_t *w;
+
+        /* Directory mode specified, but path is not a directory. */
+        if((modes & EFILE_MODE_DIRECTORY) && !handle_has_file_attributes(handle, FILE_ATTRIBUTE_DIRECTORY)) {
+            CloseHandle(handle);
+            return ENOTDIR;
+        }
+
+        /* We could check if handle has read and write permissions that
+         * correspond to the modes parameter here. However, how to do this
+         * does not seem to be very well documented and is not necessary
+         * for the current usage. */
+
+        w = (efile_win_t*)enif_alloc_resource(nif_type, sizeof(efile_win_t));
+        w->handle = handle;
```

Comment:
You need to mark the resource as having been created from a file descriptor, and store the original fd so it can be `_close`d. The documentation says we can't call `CloseHandle` on these.

Reply:
Thanks again, I will fix that.

Reply:
Fixed: 62a29aa, 9472dbe